doc/how_to/custom_components/esm/panes.md

```diff
@@ -0,0 +1,426 @@
+# Create Panes with ReactiveHTML
+
+In this guide we will show you how to efficiently implement custom panes using `JSComponent`, `ReactComponent` and `AnyWidgetComponent` to get input from the user.
+
+## Creating a ChartJS Pane
+
+This example will show you the basics of creating a [ChartJS](https://www.chartjs.org/docs/latest/) pane.
+
+```{pyodide}
+import panel as pn
+import param
+from panel.custom import JSComponent
+
+
+class ChartJSComponent(JSComponent):
```

Comment:
When trying to convert to `ReactComponent` there is an unexpected error in the console. Its unexpected because I don't use `useRef`. **Is it a bug in Panels implementation?**

```python
import panel as pn
import param
from panel.custom import ReactComponent


class ChartJSComponent(ReactComponent):
    object = param.Dict()

    _esm = """
import {
  BarController,
  LinearScale,
  BarElement,
  TimeScale,
  Tooltip,
} from 'https://esm.sh/chart.js';
import 'https://esm.sh/chartjs-adapter-date-fns';
import { enUS } from 'https://esm.sh/date-fns/locale';
import { ReactChart } from "https://esm.sh/chartjs-react"

ReactChart.register(BarController, LinearScale, BarElement, TimeScale, Tooltip);
export function render({ model }) {
    return (
        <ReactChart type={ model.object.type } data={ model.object.data } />
    );
};
"""


def data(chart_type="line"):
    return {
        "type": chart_type,
        "data": {
            "labels": ["January", "February", "March", "April", "May", "June", "July"],
            "datasets": [
                {
                    "label": "Data",
                    "backgroundColor": "rgb(255, 99, 132)",
                    "borderColor": "rgb(255, 99, 132)",
                    "data": [0, 10, 5, 2, 20, 30, 45],
                }
            ],
        },
        "options": {
            "responsive": True,
            "maintainAspectRatio": False,
        },
    }


chart_type = pn.widgets.RadioBoxGroup(
    name="Chart Type", options=["bar", "line"], inline=True
)
chart = ChartJSComponent(
    object=pn.bind(data, chart_type), height=400, sizing_mode="stretch_width"
)
pn.Column(chart_type, chart).servable()
```

![image](https://github.com/user-attachments/assets/6cdfca87-e508-4869-8663-d607eb3a4baf)

Reply:
No, you just had to make sure to specify compatible versions of Chart.js and the React library.

Reply:
How would I know what those are?

Reply:
By reading the docs issues for Chart.js and the React wrapper, really nothing at all to do with Panel.

Reply:
Ok. Its probably a bit more general than Chart.js. I believe I've seen it when using other libraries too.

Reply:
Yes, the JS world is finicky particularly in regard to React version compatibility but that's not something we can help them with beyond highlighting it in the documentation (which we should do).